packages/commonwealth/client/scripts/hooks/cosmos/useProposalMetadata.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useRef, useState } from 'react';
+import axios from 'axios';
+import * as isIPFS from 'is-ipfs';
+import { IApp } from 'state';
+import type { AnyProposal } from 'models/types';
+import { CosmosProposalV1 } from 'controllers/chain/cosmos/gov/v1/proposal-v1';
+
+interface Props {
+  app: IApp;
+  proposal: AnyProposal;
+}
+
+interface Response {
+  metadata: any;
+}
+
+export const useProposalMetadata = ({ app, proposal }: Props): Response => {
+  const [metadata, setMetadata] = useState();
+  const hasFetchedDataRef = useRef(false);
+
+  useEffect(() => {
+    if (
+      proposal instanceof CosmosProposalV1 &&
+      proposal?.data?.metadata &&
+      !hasFetchedDataRef.current &&
+      !proposal.data.title // TODO: remove this check if we want to show all metadata.
+      //See https://github.com/hicommonwealth/commonwealth/issues/4187
+    ) {
+      const fileURI = proposal.data.metadata.replace('ipfs://', '');
+      const isIPFSFile = isIPFS.cid(fileURI);
+      const metadataUri = isIPFSFile
+        ? `${app.serverUrl()}/ipfsProxy?hash=${fileURI}`
+        : fileURI;
```

Comment:
@jnaviask I am allowing it to fetch any URL put into the proposal.metadata on-chain object. I wonder if this is something we should disable until the need comes? It is easy to see somebody using a regular cloud bucket, or even a centralized database API endpoint, who knows.

I just wonder if it is a good idea to keep this open-ended.

Reply:
@mhagel let's disable arbitrary fetching until we encounter a non-IPFS URL. Too much risk.

Reply:
Sounds good. I made a new [issue](https://github.com/hicommonwealth/commonwealth/issues/4233) and disabled non-IPFS for now.